extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/FileSystemStaticHandler.java

```diff
@@ -0,0 +1,219 @@
+package io.quarkus.vertx.http.runtime.devmode;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.locks.ReentrantLock;
+
+import io.quarkus.fs.util.ZipUtils;
+import io.vertx.core.Handler;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.http.impl.HttpUtils;
+import io.vertx.core.http.impl.MimeMapping;
+import io.vertx.core.net.impl.URIDecoder;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.impl.Utils;
+
+/**
+ * A Handler to serve static files from jar files or from a local directory.
+ */
+public class FileSystemStaticHandler implements Handler<RoutingContext>, Closeable {
```

Comment:
Doesn't Vert.x already have a static handler?

Reply:
Yes, vert.x already has `StaticFilterHandlerImpl`.
This one can't however serve files from a specific jar file, only from the whole classpath.
Also, the order in which the files are served can not be specified. For example, here we need to first serve files from disk (the user provided overrides), and only then use the webjar content.

I have not seen how this can be achieved with the default `StaticFilterHandlerImpl`. Any ideas?

Reply:
I don't believe it can be achieved with the default static filter handler. As mentioned, it serves from the full classpath, in an unscoped fashion.